### DISCUSSION TITLE
How can I switch from FP16 to FP32 in the code to avoid the warning?

### DISCUSSION THREAD

**Discussion by callmefredcom** [openai/whisper#301] [Q&A] [answered]
Using Whisper in Python 3.7 (via PyCharm) on my Mac running Catalina (version 10.15.7), I get this warning:

**UserWarning: FP16 is not supported on CPU; using FP32 instead**

What should I adapt in the code to force FP32 and avoid this warning?

**Comment by jongwook** [comment, marked as answer]
`fp16=False` in Python and `--fp16 False` in the command line. I realize this is a nuisance and should make fp32 the default on CPU without warning.

**Reply by wqqinbox** [reply]
Excuse me, could you please let me know where to put the parameter "fp16=False" in Python? Any example code? Thanks.

**Reply by lsrami** [reply]
whisper recoder.wav --model tiny  --device cpu --fp16 False

**Reply by AliElshenawy** [reply]
don't work, plz illustrate more the reason and it's solution 
error " FileNotFoundError: [WinError 2] The system cannot find the file specified"

**Reply by JohnZolton** [reply]
Using the second example in the readme:

`options = whisper.DecodingOptions(language= 'en', fp16=False)`
 `result = whisper.decode(model, mel, options)`
`print(result.text`

**Reply by wqqinbox** [reply]
Thanks for the reply. I've sorted myself. Actually I used model.transcribe() function and if I add the parameters in the following way, it works well.

`result = model.transcribe(audioPath, fp16=False, language='English')`

**Reply by nalibjchn** [reply]
> Using the second example in the readme:
> 
> `options = whisper.DecodingOptions(language= 'en', fp16=False)` `result = whisper.decode(model, mel, options)` `print(result.text`

do you know what the difference is for the two examples

**Reply by MubashirAhmad12** [reply]
> don't work, plz illustrate more the reason and it's solution error " FileNotFoundError: [WinError 2] The system cannot find the file specified"

Check that the ffmpeg is correctly installed in your system.

**Comment by SugarDaddymomo** [comment]
So I faced both of the issues
1. UserWarning: FP16 is not supported on CPU; using FP32 instead
2. don't work, plz illustrate more the reason and it's solution error " FileNotFoundError: [WinError 2] The system cannot find the file specified"
For the 1st one as mentioned if using CLI then simple pass --fp16 False at the end of the command 
i tested it like,
$> whisper test.mp3 --fp16 False
and for the 2nd one install ffmpeg into the system apart from pip too
what i did was 
$> choco install ffmpeg

and that's Solved both of the issues.

**Reply by Maxpaguro** [reply]
It works man! Thx a lot 
mb

**Reply by Uzeyir36375** [reply]
Brother can you help me please? I am still facing the same issue.

**Reply by SugarDaddymomo** [reply]
> Brother can you help me please? I am still facing the same issue.
> 

Yes are you still facing the issue?

**Reply by Uzeyir36375** [reply]
Yes, Brother

**Reply by SugarDaddymomo** [reply]
Can you share the system configurations and what versions are you using and also the complete error log.

**Reply by Abzal171** [reply]
Hello,i am another guy,but i have problems with this.
Problem 1: UserWarning: FP16 is not supported on CPU; using FP32 instead.
Problem 2: [WinError 2] Не удается найти указанный файл(not kind be find named file or smth like that).
Please help me.I need it for diplom work.

**Reply by Abzal171** [reply]
System configuration: cpu Ryzen 5 5600,gpu RX 470, ram 32gb ddr4 3666mhz, os Windows 11, idle pycharm 2023.3.3

**Comment by VenomShot-real** [comment]
id also like a solution, windows and i'm using python
